sdk/core/azure-core/src/strings.cpp

```diff
@@ -73,18 +73,14 @@ namespace Azure { namespace Core { namespace Details {
     auto result = std::string(src);
     for (auto i = result.begin(); i < result.end(); i++)
     {
-      *i = c_LocaleInvariantUppercaseTable[static_cast<unsigned char>(*i)];
-      if (*i >= 'A' && *i <= 'Z')
-      {
-        *i += 32;
-      }
+      *i = ToLower(static_cast<unsigned char>(*i));

```

Comment:
I'm not sure whether we should do this.

The original intention (when there wasn't this `if` clause here) is to make full use of processor instruction-level parallelism so that this function can be really fast.

But after your change, we're leaving this up to compiler optimization. I don't think it's a good idea.